schema/me/conversation/send_message_mutation.js

```diff
@@ -0,0 +1,82 @@
+import impulse from "lib/loaders/impulse"
+import gravity from "lib/loaders/gravity"
+import { GraphQLString, GraphQLNonNull } from "graphql"
+import { mutationWithClientMutationId, cursorForObjectInConnection } from "graphql-relay"
+import { ConversationType, MessageEdge } from "./index"
+
+const { IMPULSE_APPLICATION_ID } = process.env
+
+const getImpulseToken = gravityToken => {
+  return gravity.with(gravityToken, { method: "POST" })("me/token", {
+    client_application_id: IMPULSE_APPLICATION_ID,
+  })
+}
+
+export default mutationWithClientMutationId({
+  name: "SendConversationMessageMutation",
+  description: "Appending a message to a conversation thread",
+  inputFields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The id of the conversation to be updated",
+    },
+    from: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The email address of the message sender",
+    },
+    body_text: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    reply_to_message_id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The message being replied to",
+    },
+  },
+  outputFields: {
+    conversation: {
```

Comment:
This is the conversation payload, at a glance I would say that at least `last_message` and `last_message_at` are different. We don’t really care about those on our current use, so the requirement may just be a ‘graph consistency’ requirement. I can try to look at this more after the first beta.

```
5:30:12 PM web.1     |  { id: 701,
5:30:12 PM web.1     |    inquiry_id: '595abe588b3b811cd392fa8c',
5:30:12 PM web.1     |    from_id: '547c73d57261692d83d70000',
5:30:12 PM web.1     |    from_type: 'User',
5:30:12 PM web.1     |    from_name: 'Eloy Durán',
5:30:12 PM web.1     |    from_email: 'eloy.de.enige@gmail.com',
5:30:12 PM web.1     |    to_id: '57c715b7a09a6710fb0005ba',
5:30:12 PM web.1     |    to_type: 'Partner',
5:30:12 PM web.1     |    to_name: 'Galerie Geraldine Banier',
5:30:12 PM web.1     |    intercepted: false,
5:30:12 PM web.1     |    initial_message: 'For my baby’s bedroom.',
5:30:12 PM web.1     |    created_at: '2017-07-03T22:00:08.830Z',
5:30:12 PM web.1     |    radiation_conversation_id: 1457,
5:30:12 PM web.1     |    last_message_at: '2017-07-06T15:21:22.753Z',
5:30:12 PM web.1     |    purchase_request: false,
5:30:12 PM web.1     |    items: 
5:30:12 PM web.1     |     [ { item_id: '5944fd062a893a77b95cf05e',
5:30:12 PM web.1     |         item_type: 'Artwork',
5:30:12 PM web.1     |         availability: 'for sale',
5:30:12 PM web.1     |         list_price_min: 2068,
5:30:12 PM web.1     |         list_price_max: 2068 } ],
5:30:12 PM web.1     |    _embedded: 
5:30:12 PM web.1     |     { last_message: 
5:30:12 PM web.1     |        { id: 1302,
5:30:12 PM web.1     |          radiation_message_id: 1620,
5:30:12 PM web.1     |          radiation_created_at: '2017-07-06T15:21:22.753Z',
5:30:12 PM web.1     |          from_email_address: 'eloy.de.enige@gmail.com',
5:30:12 PM web.1     |          snippet: '33' } },
```